test/fixtures/manifests/failure.pp

```diff
@@ -0,0 +1,37 @@
+# @summary A class to simulate failures
+#
+# @param fail
+#   The ways to trigger a failure
+class testing (
+  Array[Enum['exec']] $fail = [],
+) {
+  $fail.each |$failure| {
+    case $failure {
+      'exec': {
+        $command = '/tmp/failing-command'
```

Comment:
I don't like this hardcoded path, but didn't find an easy way yet. Not that I looked very hard.